src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -435,33 +437,38 @@ Status GcsActorManager::RegisterActor(const ray::rpc::RegisterActorRequest &requ
   }
 
   // The backend storage is supposed to be reliable, so the status must be ok.
-  RAY_CHECK_OK(gcs_table_storage_->ActorTable().Put(
-      actor->GetActorID(), *actor->GetMutableActorTableData(),
+  RAY_CHECK_OK(gcs_table_storage_->ActorTaskSpecTable().Put(
+      actor->GetActorID(), actor->GetCreationTaskSpecification().GetMessage(),
```

Comment:
If an actor is dead here, there should be no problem right? (I cannot really think of issues here, but I'd like to know your thought!)

Reply:
There should be a problem. If the actor is dead, we may override the actor state. I think I need to add a judgment on whether the actor is dead before put ActorTable, thanks.